beacon_chain/types/src/validator_record.rs

```diff
@@ -44,6 +45,46 @@ impl ValidatorRecord {
     }
 }
 
+impl TreeHash for ValidatorRecord {
+    /* python sample code:
+        def hash_validator_record(val):
+            return hash(val.pubkey.to_bytes(32, 'big') + val.withdrawal_shard.to_bytes(2, 'big') + \
+                val.withdrawal_address + val.randao_commitment + val.balance.to_bytes(16, 'big') + \
+                val.start_dynasty.to_bytes(8, 'big') + val.end_dynasty.to_bytes(8, 'big'))
+    */
+    fn tree_hash(&self) -> Vec<u8> {
+        // the serialized fields, to be hashed, should add up to 118 bytes in length.
+        // allocating it once here
+        let mut ssz = Vec::with_capacity(118);
```

Comment:
It would be nice to break `118` out into a constant like we've been doing [here](https://github.com/sigp/lighthouse/blob/77ac5f77bb693aa119952d033a01ebf5ec659c2a/beacon_chain/types/src/beacon_block.rs#L6-L15).

It might just be `SSZ_VALIDATOR_RECORD_LENGTH`, because it's fixed-size.

These structs keep changing as the spec has been updating, so I've also been adding [tests](https://github.com/sigp/lighthouse/blob/77ac5f77bb693aa119952d033a01ebf5ec659c2a/beacon_chain/types/src/beacon_block.rs#L121-L129) to ensure that SSZ length isn't missed during a spec update.